website/static/js/quickProjectSearchPlugin.js

```diff
@@ -0,0 +1,551 @@
+                    self.getFamilyName(1, node) + ', and ' +
+                    self.getFamilyName(2, node)
+            }
+            else {
+                return self.getFamilyName(0, node) + ', ' +
+                    self.getFamilyName(1, node) + ', ' +
+                    self.getFamilyName(2, node) + ' + ' + (numContributors - 3)
+            }
+
+        };
+
+         // Formats date for display
+        self.formatDate = function (node) {
+            return new $osf.FormattableDate(node.attributes.date_modified).local
+        };
+
+
+        // Shifts nodes back to master node list (from displayedNodes or nonMatchingNodes)
+        self.restoreToNodeList = function (missingNodes) {
+            for (var i = 0; i < missingNodes.length ; i++) {
+                self.nodes().push(missingNodes[i])
+            }
+        };
+
+        self.sortAlphabeticalAscending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.title.toUpperCase();
+                var B = b.attributes.title.toUpperCase();
+                return (A < B) ? -1 : (A > B) ? 1 : 0;
+            });
+            self.sortState('alphaAsc');
+        };
+
+        self.sortAlphabeticalDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.title.toUpperCase();
+                var B = b.attributes.title.toUpperCase();
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('alphaDesc');
+        };
+
+        self.sortDateAscending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A < B) ? -1 : (A > B) ? 1 : 0;
+            });
+            self.sortState('dateAsc')
+        };
+
+        self.sortDateDescending = function () {
+            self.nodes().sort(function(a,b){
+                var A = a.attributes.date_modified;
+                var B = b.attributes.date_modified;
+                return (A > B) ? -1 : (A < B) ? 1 : 0;
+            });
+            self.sortState('dateDesc');
+        };
+
+        // Sorts nodes depending on current sort state.
+        self.sortBySortState = function () {
+            if (self.sortState() === 'alphaAsc') {
+                self.sortAlphabeticalAscending()
+            }
+            else if (self.sortState() === 'alphaDesc') {
+                self.sortAlphabeticalDescending()
+            }
+            else if (self.sortState() === 'dateAsc') {
+                self.sortDateAscending()
+            }
+            else {
+                self.sortDateDescending()
+            }
+        };
+
+        // For xs screen
+        self.sortFieldGivenDirection = function(){
+            var dropdown = document.getElementById('sortDropDown');
+            var fieldSort = dropdown.options[dropdown.selectedIndex].value;
```

Comment:
Oh okay I remember now. Yes you are right, it is a bit more nuanced though, the point is to not mix DOM manipulation with mithril in certain ways. In the examples that you have you are still doing DOM manipulation with the vanilla js api, while not using a mithril way. So it's not a huge improvement over jQuery. It's a matter of architecture as well if mithril is not manipulating the element but just producing it then jQuery would be fine but if both of them manipulate it there may be problems. This might need further explanation, find me if you'd like to talk more.